### PR TITLE
PHP7.4 - Utilize 7.4 and clean ups

### PR DESCRIPTION
**Pull request by SpacePossum**

### CODE REVIEW

File: src/AbstractPhpdocToTypeDeclarationFixer.php [hunk 1]
```diff
@@ -31,10 +31,7 @@
  */
 abstract class AbstractPhpdocToTypeDeclarationFixer extends AbstractFixer implements ConfigurableFixerInterface
 {
-    /**
-     * @var string
-     */
-    private $classRegex = '/^\\\\?[a-zA-Z_\\x7f-\\xff](?:\\\\?[a-zA-Z0-9_\\x7f-\\xff]+)*$/';
+    private string $classRegex = '/^\\\\?[a-zA-Z_\\x7f-\\xff](?:\\\\?[a-zA-Z0-9_\\x7f-\\xff]+)*$/';
```
Comment (keradus) [inline src/AbstractPhpdocToTypeDeclarationFixer.php:34]: omg, diff is insane.
did you automated those changes? maybe we should have a rule for it?

Comment (SpacePossum) [reply src/AbstractPhpdocToTypeDeclarationFixer.php:34]: I've done these manually, but I think there is a (very) risky rule do this :sweat_smile:  , will take a look :+1:

Comment (keradus) [reply src/AbstractPhpdocToTypeDeclarationFixer.php:34]: maybe plan it as separated ticket , and we pick it one day, if prioritised

also, don't hesitate to split PR next time to smaller pieces, like "adding types" or "dropping dedicated providers for 74", now it looks like all in one place, scary to review

Comment (SpacePossum) [reply src/AbstractPhpdocToTypeDeclarationFixer.php:34]: >maybe plan it as separated ticket , and we pick it one day, if prioritised

:+1:  adding more types is not high on my list for now TBH

>split PR next time to smaller pieces

It is a split of from https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/6280 , but we can split into smaller bits if it is of value. However most changes are in the unittest so will get tested anyways by the CI.

Comment (keradus) [reply src/AbstractPhpdocToTypeDeclarationFixer.php:34]: let's then give time to review this PR and not rush with merge, pls :D

Comment (SpacePossum) [reply src/AbstractPhpdocToTypeDeclarationFixer.php:34]: ok :+1:  I'll try to split off the PR in some smaller batches, however I don't want to loose momentum here as the reviews capacity lately seems pretty limited even on the smaller PR ones, so lets keep focus on the proposed changes and not the size of the diff/PR

**Pull request closed without merging**